VideoLocker/src/main/java/org/edx/mobile/task/SetThreadReadTask.java

```diff
@@ -11,22 +11,15 @@
 
 public class SetThreadReadTask extends Task<DiscussionThread> {
     private final DiscussionThread thread;
-    private final boolean read;
 
     public SetThreadReadTask(@NonNull Context context,
-                             @NonNull DiscussionThread thread, boolean read) {
+                             @NonNull DiscussionThread thread) {
         super(context);
         this.thread = thread;
-        this.read = read;
     }
 
     @Override
     public DiscussionThread call() throws RetroHttpException {
-        return environment.getDiscussionAPI().setThreadRead(thread, read);
-    }
-
-    @Override
-    protected void onSuccess(DiscussionThread discussionThread) {
-        EventBus.getDefault().post(new DiscussionThreadUpdatedEvent(discussionThread));
+        return environment.getDiscussionAPI().setThreadRead(thread, true);
```

Comment:
> My suggestion was to retain the GetThreadTask name, and just replace it with the actual functionality of SetThreadReadTask.

Its semantically wrong to name something what it is not.

> The GET method definitions can also be removed from the API definitions, as we're unlikely to need to use them in the future. That's up to you though.

It might come in handy in the future since we won't be wanting to set a thread's read state, if only what we want is to GET the thread details.